regcomp.c

```diff
@@ -1433,12 +1436,12 @@ S_scan_commit(pTHX_ const RExC_state_t *pRExC_state, scan_data_t *data,
 	else { /* float */
 	    data->substrs[1].max_offset = (l
                           ? data->last_start_max
-                          : (data->pos_delta > SSize_t_MAX - data->pos_min
-					 ? SSize_t_MAX
+                          : (data->pos_delta > OPTIMIZE_INFTY - data->pos_min
+					 ? OPTIMIZE_INFTY
 					 : data->pos_min + data->pos_delta));
 	    if (is_inf
-		 || (STRLEN)data->substrs[1].max_offset > (STRLEN)SSize_t_MAX)
-		data->substrs[1].max_offset = SSize_t_MAX;
+		 || (STRLEN)data->substrs[1].max_offset > (STRLEN)OPTIMIZE_INFTY)
```

Comment:
Note that this test became insane as part of ea3daa5d5f6, which changed definition of max_offset from U32 to SSize_t, and our infty value from I32_MAX to SSize_t_MAX. We should fix that in a separate commit though, probably by moving the remaining `is_inf` check to be another tertiary above.